### ISSUE TITLE
Capybara::Webkit::ClickFailed

### ISSUE THREAD

**Issue by makotoshimoda** [thoughtbot/capybara-webkit#798]
Hi, I`m having a problem to select a Radio button. Capybara is saying that the same element are overlapping himself.

Capybara::Webkit::ClickFailed: Failed to click element /html/body/div[1]/div[3]/form[@id='form']/div[1]/div[@id='pr']/div/div[1]/label/input[@id='X'] because of overlapping element /html/body/div[1]/div[3]/form[@id='form']/div[1]/div[@id='pr']/div/div[1]/label/input[@id='X'] at position 0, 0;

I can find the element, but can t click.
Thanks for the help.

**Comment by jferris** [comment]
Could you attach a screenshot (using `save_screenshot`) of the page when trying to click on the element?

Also, the output from `puts page.driver.version` might be helpful.

**Comment by schmijos** [comment]
The problem also exists for me:

```
Capybara::Webkit::ClickFailed:
       Failed to click element /html/body/main/div/form[@id='new_wizard_steps_building']/div[1]/div/div[3]/div/div[1]/div[1]/div[@id='search-clear-cross'] because of overlapping element /html/body/main/div/form[@id='new_wizard_steps_building']/div[1]/div/div[3]/div/div[1]/div[1]/div[@id='search-clear-cross'] at position 1105, 280;
       A screenshot of the page at the time of the failure has been written to /var/folders/tk/x859v0090f97200xmtd4x1yw0000gn/T/click_failed_R13590.png
```

<img width="487" alt="screen shot 2016-01-26 at 20 53 16" src="https://cloud.githubusercontent.com/assets/245443/12592949/f89e542c-c46e-11e5-8cff-9603d8295a5c.png">

```
<div class="input-custom-group">
  <div class="input-custom-group-addon search">
  <object type="image/svg+xml" data="http://127.0.0.1:49634/assets/new/icon-suchen-535908b7d47073acbd35b2e13a4cc9fdb8dc156472eebb91b8e35bc4285ccca5.svg">Your browser does not support SVG</object>
  </div>
  <input class="form-control space-bottom-20" placeholder="z.B. Adresse" type="text" name="wizard_steps_building[search_string]" id="wizard_steps_building_search_string">
  <div class="input-custom-group-addon cross" id="search-clear-cross" style="display: block;">
    <object type="image/svg+xml" data="http://127.0.0.1:49634/assets/new/icon-schliessen-9c023d8d755c3c1c595e6afacf38ea6831803cddacc2ac4d5f1caec725ef6039.svg">Your browser does not support SVG</object>
  </div>
</div>
```

The cross should be clicked like that:

    find(:xpath, '//*[@id="search-clear-cross"]').click

The output of `page.driver.version` is:

    Capybara: 2.6.0
    capybara-webkit: 1.8.0
    Qt: 5.5.1
    WebKit: 538.1
    QtWebKit: 5.5.1

Maybe it's important to mention that at the beginning of running the tests the following warning appears:

```
2016-01-26 20:46:21.671 webkit_server[13590:54344] Cannot find executable for CFBundle 0x7fb56b503bf0 </Library/Internet Plug-Ins/Disabled Plug-Ins> (not loaded)
objc[13590]: Class MacCocoaSocketServerHelperRtc is implemented in both /Library/Internet Plug-Ins/googletalkbrowserplugin.plugin/Contents/MacOS/googletalkbrowserplugin and /Library/Internet Plug-Ins/o1dbrowserplugin.plugin/Contents/MacOS/o1dbrowserplugin. One of the two will be used. Which one is undefined.
```

**Comment by jferris** [comment]
That's definitely a strange error message. Trying to follow through how capybara-webkit handles this, it looks like it found the element at the expected position in JavaScript, and then the click test in Qt failed. I'm not sure I've seen that before.

Screenshots from capybara-webkit usually have a mouse cursor drawn where we were trying to click, and I don't see one in the screenshot you attached. Was that screenshot from the failure you pasted?

**Comment by schmijos** [comment]
Yes, what I think is relevant of the screenshot. And yes, there's **no** cursor visible on the screenshot.

As a **workaround** I now use the following test which works:

    find(:xpath, '//*[@id="search-clear-cross"]').trigger('click')

*(Just be aware that this doesn't test exactly the same thing.)*

**Comment by RyanQuey** [comment]
I'm having a similar problem, where capybara WebKit is raising an error that make this I cannot click on something because of overlapping elements that do not overlap in either the development or production environments. 

Is there any rhyme or reason behind what elements overlap?

Also, why does the previous workaround using `.trigger('click')` work?

**Comment by mhoran** [comment]
@RyanQuey it's tough to tell what may be causing the overlap, but it can be something to do with the size of the viewport, or just unexpected positioning of items that look different due to slight rendering differences. You may also be using a library that adds other content to the page transparently.

The `.trigger('click')` mechanism works because it doesn't actually click the element; it only sends a JavaScript click event. We've implemented virtual click support by moving a hidden cursor to the position of an element before sending a click event through your operating system window manager, as the JavaScript trick doesn't work with certain elements like the file upload field. It also doesn't trigger all the associated callbacks that a real click triggers.

**Comment by RyanQuey** [comment]
OK, got it, thanks for your response. It's helpful to get more details regarding what is happening behind the scenes. 

Hopefully in the future we'll figure out a way to resolve this overlapping issue.

**Comment by prusswan** [comment]
I am getting a variant of this issue, which is related to an element in a dropdown menu on hover. The dropdown is implemented using some non-standard javascript (with `href=javascript:void(2)` etc) and stays open if it is clicked on. But since webkit is not able to open the dropdown with any means, I had to resort to `trigger('click')` to avoid this issue, and even so the results are pretty unreliable. Had to switch to poltergeist for now.

**Comment by kannans** [comment]
I have got the same problem when I upgrade the qt version `qt@5.5`. Later I have removed `qt@5.5` and installed old version of  `qt@4`. It get resolved my problem.

**Comment by twalpole** [comment]
Closing since capybara-webkit is behaving as expected -- when an element is overlapping another element the user couldn't click the element underneath so neither will capybara-webkit.
